homeassistant/components/roborock/__init__.py

```diff
@@ -43,19 +47,27 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         for device in device_map.values()
     }
     network_results = await asyncio.gather(
-        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values())
+        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values()),
+        return_exceptions=True,
     )
-    network_info = {
-        device.duid: result
-        for device, result in zip(device_map.values(), network_results)
-        if result is not None
-    }
+    network_info = {}
+    for device, result in zip(device_map.values(), network_results):
+        if result is not None and not isinstance(result, RoborockException):
+            network_info[device.duid] = result
+        else:
+            _LOGGER.warning(
+                "Failed to connect to get networking information about %s", device.duid
+            )
+            if isinstance(result, RoborockException):
+                _LOGGER.exception(result)
```

Comment:
I feel like this can be simplified. I don't directly see how, but currently if result is None, result gets a second check at line 61 if None isn't a RoborockException and that feels strange

Reply:
I could switch it around?


```
        if result is None or isinstance(result, RoborockException):
            _LOGGER.warning(
                "Failed to connect to get networking information about %s", device.duid
            )
            if isinstance(result, RoborockException):
                _LOGGER.exception(result)
        else:
            network_info[device.duid] = result
```

Not sure if that is really any better though. I could also repeat code/ just make a seperate warning for roborockexception and do an if/elif.

Let me think about this a bit more and get back to it

Reply:
I made a change that is at least easier to follow and only makes each check once